### DISCUSSION TITLE
Navigate to all FHIR resource types

### DISCUSSION THREAD

**Discussion by codyebberson** [medplum/medplum#1517] [Ideas]
# Navigate to all FHIR resource types

One of the current limitations of https://app.medplum.com is that it is difficult to find FHIR resource types that are not listed in the left navigation bar.  Any resource type **can** be added to the left bar, but that is an extra configuration step, and a heavier lift than necessary.

Instead, advanced Medplum users know to type the resource type directly into the URL bar, such as https://app.medplum.com/MedicationRequest or https://app.medplum.com/ImagingStudy -- also not ideal.

This discussion is about how to display the full list of FHIR resources to the user?

I am not a UX expert, so these are just some options that seem reasonable to me in the short term.

Summary:

1. Link to a page that lists all resource types
2. Autocomplete box inside the left nav panel
3. Extend the header autocomplete to include resource types

## 1. Link to a page that lists all resource types

This is probably the most straightforward option, although perhaps not the most elegant.  The idea is to include a simple link (either in the left nav or in the top-right popup menu) to a big page that lists all resource types.

For example, link in the left nav:

![image](https://user-images.githubusercontent.com/749094/218351896-3b4c4d79-7cc2-49c1-bed8-669004b6ebe3.png)

The page itself could either be a big alphabetical listing, or perhaps grouped by "module" or "level" such as on the [FHIR Resources page](https://hl7.org/fhir/): 

![image](https://user-images.githubusercontent.com/749094/218352008-49f7e0f0-681f-412a-bd81-0dcf12a4bc2d.png)

## 2. Autocomplete box inside the left nav panel

Similar to the [Mantine "Navbar with search" example](https://ui.mantine.dev/component/navbar-search):

![image](https://user-images.githubusercontent.com/749094/218352145-60c7a027-10b5-4ca2-a634-7f30df3c32c8.png)

## 3. Extend the header autocomplete to include resource types

Currently, the header autocomplete searches for "Patient" and "ServiceRequest" that matches the search string.

In general, we should probably search for more resource types (?)

But we could also make it a more rich display.

Example, the Algolia search widget has a lot of complex functionality:

![image](https://user-images.githubusercontent.com/749094/218352435-e7fe65fe-e70a-47fa-bd7e-27eee990093b.png)

Example, StackOverflow has a multi column popup:

![image](https://user-images.githubusercontent.com/749094/218352376-ab66e858-6d0c-4d1a-a2ff-22d1cfdef0e5.png)

Example: Mantine [mega menu example](https://ui.mantine.dev/component/header-mega-menu#)

![image](https://user-images.githubusercontent.com/749094/218352611-78312114-8c81-4888-9031-97c00504197c.png)

**Comment by rahul1** [comment]
Left Nav Bar [Option 2] seems most consistent with our current model. Our current users are already used to looking for resources there, so adding the autocomplete box there makes the most sense to me

**Comment by codyebberson** [comment]
Fixed in https://github.com/medplum/medplum/pull/1844
